--- a/tools/python/xen/xm/create.py
+++ b/tools/python/xen/xm/create.py
@@ -639,6 +639,14 @@
     else:
         return s
 
+def abspath(file, dir):
+    if file[0] == '/':
+        return file
+
+    import xen.util.auxbin
+    path = xen.util.auxbin.root()
+    return os.path.join(path, dir, file)
+
 def configure_image(vals):
     """Create the image config.
     """
@@ -646,11 +654,11 @@
         return None
     config_image = [ vals.builder ]
     if vals.kernel:
-        config_image.append([ 'kernel', os.path.abspath(vals.kernel) ])
+        config_image.append([ 'kernel', abspath(vals.kernel, 'boot') ])
     if vals.ramdisk:
-        config_image.append([ 'ramdisk', os.path.abspath(vals.ramdisk) ])
+        config_image.append([ 'ramdisk', abspath(vals.ramdisk, 'boot') ])
     if vals.loader:
-        config_image.append([ 'loader', os.path.abspath(vals.loader) ])
+        config_image.append([ 'loader', abspath(vals.loader, 'boot') ])
     if vals.cmdline_ip:
         cmdline_ip = strip('ip=', vals.cmdline_ip)
         config_image.append(['ip', cmdline_ip])
@@ -925,7 +933,7 @@
 def configure_hvm(config_image, vals):
     """Create the config for HVM devices.
     """
-    args = [ 'device_model', 'pae', 'vcpus', 'boot', 'fda', 'fdb', 'timer_mode',
+    args = [ 'pae', 'vcpus', 'boot', 'fda', 'fdb', 'timer_mode',
              'localtime', 'serial', 'stdvga', 'isa', 'nographic', 'soundhw',
              'vnc', 'vncdisplay', 'vncunused', 'vncconsole', 'vnclisten',
              'sdl', 'display', 'xauthority', 'rtc_timeoffset', 'monitor',
@@ -937,6 +945,8 @@
     for a in args:
         if a in vals.__dict__ and vals.__dict__[a] is not None:
             config_image.append([a, vals.__dict__[a]])
+    if vals.device_model:
+        config_image.append(['device_model', abspath(vals.device_model, 'bin')])
     if vals.vncpasswd is not None:
         config_image.append(['vncpasswd', vals.vncpasswd])
 
@@ -980,10 +990,7 @@
 
     config_image = configure_image(vals)
     if vals.bootloader:
-        if vals.bootloader == "pygrub":
-            vals.bootloader = osdep.pygrub_path
-
-        config.append(['bootloader', vals.bootloader])
+        config.append(['bootloader', abspath(vals.bootloader, 'bin')])
         if vals.bootargs:
             config.append(['bootloader_args', vals.bootargs])
         else:
