--- xen-3.orig/tools/blktap/drivers/Makefile
+++ xen-3/tools/blktap/drivers/Makefile
@@ -39,11 +39,11 @@
 all: $(IBIN) qcow-util
 
 blktapctrl: blktapctrl.c
-	$(CC) $(CFLAGS) -o blktapctrl $(LIBS) blktapctrl.c
+	$(CC) $(CFLAGS) -o blktapctrl $(LDFLAGS) $(LIBS) blktapctrl.c
 
 tapdisk: $(BLK-OBJS) tapdisk.c
 	$(CC) $(CFLAGS) -o tapdisk $(BLK-OBJS) tapdisk.c \
-		$(AIOLIBS) $(LIBS)
+		$(AIOLIBS) $(LDFLAGS) $(LIBS)
 
 .PHONY: qcow-util
 qcow-util: img2qcow qcow2raw qcow-create
--- xen-3.orig/tools/console/Makefile
+++ xen-3/tools/console/Makefile
@@ -20,11 +20,11 @@
 	$(RM) client/*.o daemon/*.o
 
 xenconsoled: $(patsubst %.c,%.o,$(wildcard daemon/*.c))
-	$(CC) $(CFLAGS) $^ -o $@ -L$(XEN_LIBXC) -L$(XEN_XENSTORE) \
+	$(CC) $(CFLAGS) $^ -o $@ $(LDFLAGS) -L$(XEN_LIBXC) -L$(XEN_XENSTORE) \
               $(UTIL_LIBS) $(SOCKET_LIBS) -lxenctrl -lxenstore
 
 xenconsole: $(patsubst %.c,%.o,$(wildcard client/*.c))
-	$(CC) $(CFLAGS) $^ -o $@ -L$(XEN_LIBXC) -L$(XEN_XENSTORE) \
+	$(CC) $(CFLAGS) $^ -o $@ $(LDFLAGS) -L$(XEN_LIBXC) -L$(XEN_XENSTORE) \
 	      $(UTIL_LIBS) $(SOCKET_LIBS) -lxenctrl -lxenstore
 
 .PHONY: install
--- xen-3.orig/tools/misc/Makefile
+++ xen-3/tools/misc/Makefile
@@ -41,4 +41,4 @@
 	$(CC) -c $(CFLAGS) -o $@ $<
 
 xenperf: %: %.o Makefile
-	$(CC) $(CFLAGS) -o $@ $< -L$(XEN_LIBXC) -lxenctrl
+	$(CC) $(CFLAGS) -o $@ $< $(LDFLAGS) -L$(XEN_LIBXC) -lxenctrl
--- xen-3.orig/tools/xcutils/Makefile
+++ xen-3/tools/xcutils/Makefile
@@ -33,7 +33,7 @@
 build: $(PROGRAMS)
 
 $(PROGRAMS): %: %.o
-	$(CC) $(CFLAGS) $^ $(LDLIBS) -o $@
+	$(CC) $(CFLAGS) $(LDFLAGS) $^ $(LDLIBS) -o $@
 
 .PHONY: install
 install: build
