diff -Nur xen-3-3.2.1/tools/ioemu/target-i386-dm/qemu-dm.debug xen-3/tools/ioemu/target-i386-dm/qemu-dm.debug
--- xen-3-3.2.1/tools/ioemu/target-i386-dm/qemu-dm.debug	2008-05-12 13:01:56.000000000 +0200
+++ xen-3/tools/ioemu/target-i386-dm/qemu-dm.debug	2008-08-25 10:31:15.000000000 +0200
@@ -1,10 +1,11 @@
 #!/bin/sh
 
-if [ "`arch`" = "x86_64" ]; then
+if [ "`uname -m`" = "x86_64" ]; then
     LIBDIR="lib64"
 else
     LIBDIR="lib"
 fi
-echo $* > /tmp/args
-echo $DISPLAY >> /tmp/args
+TMPFILE=`mktemp`
+echo $* > $TMPFILE
+echo $DISPLAY >> $TMPFILE
 exec /usr/$LIBDIR/xen/bin/qemu-dm $*
