--- a/tools/misc/Makefile
+++ b/tools/misc/Makefile
@@ -18,13 +18,10 @@
 SUBDIRS-$(CONFIG_MINITERM) += miniterm
 SUBDIRS := $(SUBDIRS-y)
 
-INSTALL_BIN-y := xencons
-INSTALL_BIN-$(CONFIG_X86) += xen-detect
-INSTALL_BIN := $(INSTALL_BIN-y)
-
-INSTALL_SBIN-y := netfix xm xen-bugtool xen-python-path xend xenperf xsview
-INSTALL_SBIN-$(CONFIG_X86) += xenpm
-INSTALL_SBIN := $(INSTALL_SBIN-y)
+INSTALL_TARGETS-y := xencons
+INSTALL_TARGETS-y := netfix xm xen-bugtool xen-python-path xend xenperf xsview
+INSTALL_TARGETS-$(CONFIG_X86) += xenpm
+INSTALL_TARGETS := $(INSTALL_TARGETS-y)
 
 DEFAULT_PYTHON_PATH := $(shell $(XEN_ROOT)/tools/python/get-path)
 PYTHON_PATH ?= $(DEFAULT_PYTHON_PATH)
@@ -40,10 +37,8 @@
 
 .PHONY: install
 install: build
-	$(INSTALL_DIR) $(DESTDIR)$(BINDIR)
-	$(INSTALL_DIR) $(DESTDIR)$(SBINDIR)
-	$(INSTALL_PYTHON_PROG) $(INSTALL_BIN) $(DESTDIR)$(BINDIR)
-	$(INSTALL_PYTHON_PROG) $(INSTALL_SBIN) $(DESTDIR)$(SBINDIR)
+	$(INSTALL_DIR) $(DESTDIR)$(PRIVATE_BINDIR)
+	$(INSTALL_PYTHON_PROG) $(INSTALL_TARGETS) $(DESTDIR)$(PRIVATE_BINDIR)
 	set -e; for d in $(SUBDIRS); do $(MAKE) -C $$d install-recurse; done
 
 .PHONY: clean
--- a/tools/misc/lomount/Makefile
+++ b/tools/misc/lomount/Makefile
@@ -16,7 +16,7 @@
 
 .PHONY: install
 install install-recurse: build
-	$(INSTALL_PROG) $(BIN) $(SCRIPTS) $(DESTDIR)$(BINDIR)
+	$(INSTALL_PROG) $(BIN) $(SCRIPTS) $(DESTDIR)$(PRIVATE_BINDIR)
 
 .PHONY: clean
 clean:
