--- xen-3.orig/tools/Rules.mk
+++ xen-3/tools/Rules.mk
@@ -5,6 +5,8 @@
 
 include $(XEN_ROOT)/Config.mk
 
+LDFLAGS_RPATH = -Wl,-rpath,'$${ORIGIN}$(if $(1),/$(1))'
+
 XEN_XC             = $(XEN_ROOT)/tools/python/xen/lowlevel/xc
 XEN_LIBXC          = $(XEN_ROOT)/tools/libxc
 XEN_XENSTORE       = $(XEN_ROOT)/tools/xenstore
--- xen-3.orig/tools/python/setup.py
+++ xen-3/tools/python/setup.py
@@ -5,6 +5,7 @@
 XEN_ROOT = "../.."
 
 extra_compile_args  = [ "-fno-strict-aliasing", "-Werror" ]
+extra_link_args = [ "-Wl,-rpath,${ORIGIN}/../../.." ]
 
 include_dirs = [ XEN_ROOT + "/tools/libxc",
                  XEN_ROOT + "/tools/xenstore",
@@ -18,6 +19,7 @@
 
 xc = Extension("xc",
                extra_compile_args = extra_compile_args,
+               extra_link_args    = extra_link_args,
                include_dirs       = include_dirs + [ "xen/lowlevel/xc" ],
                library_dirs       = library_dirs,
                libraries          = libraries,
@@ -25,6 +27,7 @@
 
 xs = Extension("xs",
                extra_compile_args = extra_compile_args,
+               extra_link_args    = extra_link_args,
                include_dirs       = include_dirs + [ "xen/lowlevel/xs" ],
                library_dirs       = library_dirs,
                libraries          = libraries,
@@ -32,6 +35,7 @@
 
 scf = Extension("scf",
                extra_compile_args = extra_compile_args,
+               extra_link_args    = extra_link_args,
                include_dirs       = include_dirs + [ "xen/lowlevel/scf" ],
                library_dirs       = library_dirs,
                libraries          = libraries,
@@ -39,6 +43,7 @@
              
 acm = Extension("acm",
                extra_compile_args = extra_compile_args,
+               extra_link_args    = extra_link_args,
                include_dirs       = include_dirs + [ "xen/lowlevel/acm" ],
                library_dirs       = library_dirs,
                libraries          = libraries,
@@ -46,6 +51,7 @@
 
 flask = Extension("flask",
                extra_compile_args = extra_compile_args,
+               extra_link_args    = extra_link_args,
                include_dirs       = include_dirs + [ "xen/lowlevel/flask" ] + 
                                         [ "../flask/libflask/include" ],
                library_dirs       = library_dirs + [ "../flask/libflask" ],
@@ -54,6 +60,7 @@
 
 ptsname = Extension("ptsname",
                extra_compile_args = extra_compile_args,
+               extra_link_args    = extra_link_args,
                include_dirs       = include_dirs + [ "ptsname" ],
                library_dirs       = library_dirs,
                libraries          = libraries,
--- xen-3.orig/tools/ioemu/Makefile.target
+++ xen-3/tools/ioemu/Makefile.target
@@ -506,6 +506,8 @@
 VL_LDFLAGS+=-Wl,-T,$(SRC_PATH)/sparc64.ld
 endif
 
+VL_LDFLAGS+=$(call LDFLAGS_RPATH,../lib)
+
 ifdef CONFIG_WIN32
 SDL_LIBS := $(filter-out -mwindows, $(SDL_LIBS)) -mconsole
 endif
--- xen-3.orig/tools/libxc/Makefile
+++ xen-3/tools/libxc/Makefile
@@ -149,7 +149,7 @@
 	ln -sf $< $@
 
 libxenctrl.so.$(MAJOR).$(MINOR): $(CTRL_PIC_OBJS)
-	$(CC) $(CFLAGS) $(LDFLAGS) -Wl,$(SONAME_LDFLAG) -Wl,libxenctrl.so.$(MAJOR) $(SHLIB_CFLAGS) -o $@ $^ -lpthread
+	$(CC) $(CFLAGS) $(LDFLAGS) -Wl,$(SONAME_LDFLAG) -Wl,libxenctrl.so.$(MAJOR) $(SHLIB_CFLAGS) $(call LDFLAGS_RPATH) -o $@ $^ -lpthread
 
 # libxenguest
 
@@ -162,7 +162,7 @@
 	ln -sf $< $@
 
 libxenguest.so.$(MAJOR).$(MINOR): $(GUEST_PIC_OBJS) libxenctrl.so
-	$(CC) $(CFLAGS) $(LDFLAGS) -Wl,$(SONAME_LDFLAG) -Wl,libxenguest.so.$(MAJOR) $(SHLIB_CFLAGS) -o $@ $(GUEST_PIC_OBJS) -lz -lxenctrl -lpthread
+	$(CC) $(CFLAGS) $(LDFLAGS) -Wl,$(SONAME_LDFLAG) -Wl,libxenguest.so.$(MAJOR) $(SHLIB_CFLAGS) $(call LDFLAGS_RPATH) -o $@ $(GUEST_PIC_OBJS) -lz -lxenctrl -lpthread
 
 -include $(DEPS)
 
--- xen-3.orig/tools/xcutils/Makefile
+++ xen-3/tools/xcutils/Makefile
@@ -24,7 +24,7 @@
 
 PROGRAMS = xc_restore xc_save readnotes
 
-LDLIBS   = -L$(XEN_LIBXC) -L$(XEN_XENSTORE) -lxenguest -lxenctrl -lxenstore
+LDLIBS   = -L$(XEN_LIBXC) -L$(XEN_XENSTORE) -lxenguest -lxenctrl -lxenstore $(call LDFLAGS_RPATH,../lib)
 
 .PHONY: all
 all: build
--- xen-3.orig/tools/xenstat/xentop/Makefile
+++ xen-3/tools/xenstat/xentop/Makefile
@@ -25,6 +25,7 @@
 
 CFLAGS += -DGCC_PRINTF -Wall -Werror -I$(XEN_LIBXENSTAT)
 LDFLAGS += -L$(XEN_LIBXENSTAT)
+LDFLAGS += $(call LDFLAGS_RPATH,../lib)
 LDLIBS += -lxenstat $(CURSES_LIBS) $(SOCKET_LIBS)
 CFLAGS += -DHOST_$(XEN_OS)
 
--- xen-3.orig/tools/misc/Makefile
+++ xen-3/tools/misc/Makefile
@@ -44,4 +44,4 @@
 	$(CC) -c $(CFLAGS) -o $@ $<
 
 xenperf: %: %.o Makefile
-	$(CC) $(CFLAGS) -o $@ $< -L$(XEN_LIBXC) -lxenctrl
+	$(CC) $(CFLAGS) -o $@ $< -L$(XEN_LIBXC) -lxenctrl $(call LDFLAGS_RPATH,../lib)
--- xen-3.orig/tools/xentrace/Makefile
+++ xen-3/tools/xentrace/Makefile
@@ -5,6 +5,7 @@
 
 CFLAGS  += -I $(XEN_XC)
 CFLAGS  += -I $(XEN_LIBXC)
+LDFLAGS += $(call LDFLAGS_RPATH,../lib)
 
 HDRS     = $(wildcard *.h)
 OBJS     = $(patsubst %.c,%.o,$(wildcard *.c))
@@ -52,6 +53,6 @@
 	$(RM) *.a *.so *.o *.rpm $(BIN) $(LIBBIN)
 
 %: %.c $(HDRS) Makefile
-	$(CC) $(CFLAGS) -o $@ $< -L$(XEN_LIBXC) -lxenctrl
+	$(CC) $(CFLAGS) -o $@ $< $(call LDFLAGS_RPATH,../lib) -L$(XEN_LIBXC) -lxenctrl
 xentrace_%: %.c $(HDRS) Makefile
-	$(CC) $(CFLAGS) -o $@ $< -L$(XEN_LIBXC) -lxenctrl
+	$(CC) $(CFLAGS) -o $@ $< $(call LDFLAGS_RPATH,../lib) -L$(XEN_LIBXC) -lxenctrl
--- xen-3.orig/tools/console/Makefile
+++ xen-3/tools/console/Makefile
@@ -9,6 +9,7 @@
 
 CFLAGS  += -I $(XEN_LIBXC)
 CFLAGS  += -I $(XEN_XENSTORE)
+LDFLAGS += $(call LDFLAGS_RPATH,../lib)
 
 BIN      = xenconsoled xenconsole
 
@@ -21,11 +22,11 @@
 	$(RM) client/*.o daemon/*.o
 
 xenconsoled: $(patsubst %.c,%.o,$(wildcard daemon/*.c))
-	$(CC) $(CFLAGS) $^ -o $@ -L$(XEN_LIBXC) -L$(XEN_XENSTORE) \
+	$(CC) $(CFLAGS) $^ -o $@ $(LDFLAGS) -L$(XEN_LIBXC) -L$(XEN_XENSTORE) \
               $(UTIL_LIBS) $(SOCKET_LIBS) -lxenctrl -lxenstore
 
 xenconsole: $(patsubst %.c,%.o,$(wildcard client/*.c))
-	$(CC) $(CFLAGS) $^ -o $@ -L$(XEN_LIBXC) -L$(XEN_XENSTORE) \
+	$(CC) $(CFLAGS) $^ -o $@ $(LDFLAGS) -L$(XEN_LIBXC) -L$(XEN_XENSTORE) \
 	      $(UTIL_LIBS) $(SOCKET_LIBS) -lxenctrl -lxenstore
 
 .PHONY: install
--- xen-3.orig/tools/xenmon/Makefile
+++ xen-3/tools/xenmon/Makefile
@@ -19,6 +19,7 @@
 CFLAGS  += -I $(XEN_XC)
 CFLAGS  += -I $(XEN_LIBXC)
 LDFLAGS += -L $(XEN_LIBXC)
+LDFLAGS += $(call LDFLAGS_RPATH,../lib)
 
 BIN = xentrace_setmask xenbaked
 SCRIPTS = xenmon.py
--- xen-3.orig/tools/libfsimage/Rules.mk
+++ xen-3/tools/libfsimage/Rules.mk
@@ -4,6 +4,7 @@
 
 CFLAGS += -I$(XEN_ROOT)/tools/libfsimage/common/ -Werror -Wp,-MD,.$(@F).d
 LDFLAGS += -L../common/
+LDFLAGS += $(call LDFLAGS_RPATH,../..)
 
 PIC_OBJS := $(patsubst %.c,%.opic,$(LIB_SRCS-y))
 
--- xen-3.orig/tools/pygrub/setup.py
+++ xen-3/tools/pygrub/setup.py
@@ -4,11 +4,13 @@
 import sys
 
 extra_compile_args  = [ "-fno-strict-aliasing", "-Werror" ]
+extra_link_args = [ "-Wl,-rpath,${ORIGIN}/.." ]
 
 XEN_ROOT = "../.."
 
 fsimage = Extension("fsimage",
     extra_compile_args = extra_compile_args,
+    extra_link_args = extra_link_args,
     include_dirs = [ XEN_ROOT + "/tools/libfsimage/common/" ],
     library_dirs = [ XEN_ROOT + "/tools/libfsimage/common/" ],
     libraries = ["fsimage"],
--- xen-3.orig/tools/blktap/lib/Makefile
+++ xen-3/tools/blktap/lib/Makefile
@@ -10,6 +10,7 @@
 INCLUDES += -I. -I.. -I $(XEN_LIBXC) -I $(XEN_XENSTORE)
 
 LIBS     := -lz
+LIBS     += $(call LDFLAGS_RPATH)
 
 SRCS     :=
 SRCS     += xenbus.c blkif.c xs_api.c
--- xen-3.orig/tools/blktap/drivers/Makefile
+++ xen-3/tools/blktap/drivers/Makefile
@@ -25,6 +25,7 @@
 LIBS      += -lcrypto
 LIBS      += -lz
 LIBS      += -L$(XEN_XENSTORE) -lxenstore
+LIBS      += $(call LDFLAGS_RPATH,../lib)
 
 AIOLIBS   := $(LIBAIO_DIR)/libaio.a
 
--- xen-3.orig/tools/xenstore/Makefile
+++ xen-3/tools/xenstore/Makefile
@@ -33,7 +33,7 @@
 all: libxenstore.so libxenstore.a xenstored $(CLIENTS) xs_tdb_dump xenstore-control xenstore-ls
 
 xenstored: $(XENSTORED_OBJS)
-	$(CC) $(CFLAGS) $(LDFLAGS) $^ $(LOADLIBES) $(LDLIBS) -lxenctrl $(SOCKET_LIBS) -o $@
+	$(CC) $(CFLAGS) $(LDFLAGS) $^ $(LOADLIBES) $(LDLIBS) $(call LDFLAGS_RPATH,../lib) -lxenctrl $(SOCKET_LIBS) -o $@
 
 $(CLIENTS): xenstore-%: xenstore_%.o libxenstore.so
 	$(CC) $(CFLAGS) $(LDFLAGS) $< $(LOADLIBES) $(LDLIBS) -L. -lxenstore $(SOCKET_LIBS) -o $@
