--- xen-3.orig/tools/misc/xend
+++ xen-3/tools/misc/xend
@@ -33,15 +33,10 @@
 import time
 import commands
 
-xpp = os.path.join(os.path.dirname(sys.argv[0]), 'xen-python-path')
-if os.path.exists(xpp):
-    result = commands.getstatusoutput(xpp)
-    if result[0] != 0:
-        print >>sys.stderr, result[1]
-        sys.exit(1)
-    sys.path.append(result[1])
+sys.path.insert(1, sys.path[0] + '/../lib/python')
 
 from xen.xend.server import SrvDaemon
+from xen.util.auxbin import execute
 
 class CheckError(ValueError):
     pass
@@ -95,20 +90,20 @@
         f.close()
     except:
         pass
-    XENSTORED_TRACE = os.getenv("XENSTORED_TRACE")
-    cmd = "xenstored --pid-file /var/run/xenstore.pid"
-    if XENSTORED_TRACE:
-        cmd += " -T /var/log/xen/xenstored-trace.log"
-    s,o = commands.getstatusoutput(cmd)
+    args = ["--pid-file", "/var/run/xenstore.pid"]
+    if os.getenv("XENSTORED_TRACE"):
+        args.extend(["-T", "/var/log/xen/xenstored-trace.log"])
+    if os.fork() == 0:
+        execute('xenstored', args)
 
 def start_consoled():
     if os.fork() == 0:
-        os.execvp('xenconsoled', ['xenconsoled'])
+        execute('xenconsoled')
 
 def start_blktapctrl():
     if os.fork() == 0:
-        os.execvp('blktapctrl', ['blktapctrl'])
-            
+        execute('blktapctrl')
+
 def main():
     try:
         check_logging()
--- xen-3.orig/tools/misc/xm
+++ xen-3/tools/misc/xm
@@ -1,10 +1,7 @@
 #!/usr/bin/env python
 #  -*- mode: python; -*-
 import sys
-
-# add fallback path for non-native python path installs if needed
-sys.path.append('/usr/lib/python')
-sys.path.append('/usr/lib64/python')
+sys.path.insert(1, sys.path[0] + '/../lib/python')
 from xen.xm import main
 
 main.main(sys.argv)
