--- a/tools/pygrub/setup.py
+++ b/tools/pygrub/setup.py
@@ -4,11 +4,13 @@
 import sys
 
 extra_compile_args  = [ "-fno-strict-aliasing", "-Werror" ]
+extra_link_args = [ "-Wl,-rpath,${ORIGIN}/.." ]
 
 XEN_ROOT = "../.."
 
 fsimage = Extension("fsimage",
     extra_compile_args = extra_compile_args,
+    extra_link_args = extra_link_args,
     include_dirs = [ XEN_ROOT + "/tools/libfsimage/common/" ],
     library_dirs = [ XEN_ROOT + "/tools/libfsimage/common/" ],
     libraries = ["fsimage"],
