--- a/tools/blktap2/Makefile
+++ b/tools/blktap2/Makefile
@@ -1,9 +1,6 @@
 XEN_ROOT = ../..
 include $(XEN_ROOT)/tools/Rules.mk
 
-CFLAGS  += $(CFLAGS_libxenctrl)
-LDFLAGS += $(LDFLAGS_libxenctrl)
-
 SUBDIRS-y :=
 SUBDIRS-y += include
 SUBDIRS-y += lvm
--- a/tools/blktap2/vhd/Makefile
+++ b/tools/blktap2/vhd/Makefile
@@ -12,6 +12,7 @@
 CFLAGS            += -Wno-unused
 CFLAGS            += -I../include
 CFLAGS            += -D_GNU_SOURCE
+CFLAGS  += $(CFLAGS_libxenctrl)
 
 ifeq ($(CONFIG_X86_64),y)
 CFLAGS            += -fPIC
--- a/tools/blktap2/vhd/lib/Makefile
+++ b/tools/blktap2/vhd/lib/Makefile
@@ -8,9 +8,7 @@
 
 LVM-UTIL-OBJ    := $(BLKTAP_ROOT)/lvm/lvm-util.o
 
-LIBVHD-BUILD    := libvhd.a
-
-INST-DIR         = $(LIBDIR)
+LIBVHD-BUILD    := libvhd.so
 
 CFLAGS          += -Werror
 CFLAGS          += -Wno-unused
@@ -18,10 +16,12 @@
 CFLAGS          += -D_GNU_SOURCE
 CFLAGS          += -fPIC
 CFLAGS          += -g
+CFLAGS  += $(CFLAGS_libxenctrl)
 
 ifeq ($(CONFIG_Linux),y)
 LIBS            := -luuid
 endif
+LDFLAGS += $(LDFLAGS_libxenctrl)
 
 # Get gcc to generate the dependencies for us.
 CFLAGS          += -Wp,-MD,.$(@F).d
@@ -48,24 +48,19 @@
 LIB-OBJS         = $(patsubst %.c,%.o,$(LIB-SRCS))
 LIB-OBJS        += $(LVM-UTIL-OBJ)
 
-LIBVHD           = libvhd.a libvhd.so.$(LIBVHD-MAJOR).$(LIBVHD-MINOR)
+LIBVHD           = libvhd.so
 
 all: build
 
 build: $(LIBVHD-BUILD)
 
-libvhd.a: $(LIB-OBJS)
-	$(CC) $(CFLAGS) -Wl,$(SONAME_LDFLAG),$(LIBVHD-SONAME) $(SHLIB_CFLAGS) \
-		$(LDFLAGS) -o libvhd.so.$(LIBVHD-MAJOR).$(LIBVHD-MINOR) $(LIBS) $^
-	ln -sf libvhd.so.$(LIBVHD-MAJOR).$(LIBVHD-MINOR) libvhd.so.$(LIBVHD-MAJOR)
-	ln -sf libvhd.so.$(LIBVHD-MAJOR) libvhd.so
-	$(AR) rc $@ $^
+libvhd.so: $(LIB-OBJS)
+	$(CC) $(CFLAGS) $(SHLIB_CFLAGS) \
+		$(LDFLAGS) -o $@ $(LIBS) $^
 
 install: all
-	$(INSTALL_DIR) -p $(DESTDIR)$(INST-DIR)
-	$(INSTALL_DATA) $(LIBVHD) $(DESTDIR)$(INST-DIR)
-	ln -sf libvhd.so.$(LIBVHD-MAJOR).$(LIBVHD-MINOR) $(DESTDIR)$(INST-DIR)/libvhd.so.$(LIBVHD-MAJOR)
-	ln -sf libvhd.so.$(LIBVHD-MAJOR) $(DESTDIR)$(INST-DIR)/libvhd.so
+	$(INSTALL_DIR) -p $(DESTDIR)$(PRIVATE_LIBDIR)
+	$(INSTALL_DATA) $(LIBVHD) $(DESTDIR)$(PRIVATE_LIBDIR)
 
 clean:
 	rm -rf *.a *.so* *.o *~ $(DEPS) $(LIBVHD)
