--- a/tools/console/Makefile
+++ b/tools/console/Makefile
@@ -11,6 +11,7 @@
 LDLIBS += $(UTIL_LIBS)
 LDLIBS += $(SOCKET_LIBS)
 LDLIBS += -lrt
+LDFLAGS += $(call LDFLAGS_RPATH,../lib)
 
 BIN      = xenconsoled xenconsole
 
@@ -30,9 +31,7 @@
 
 .PHONY: install
 install: $(BIN)
-	$(INSTALL_DIR) $(DESTDIR)/$(SBINDIR)
-	$(INSTALL_PROG) xenconsoled $(DESTDIR)/$(SBINDIR)
 	$(INSTALL_DIR) $(DESTDIR)$(PRIVATE_BINDIR)
-	$(INSTALL_PROG) xenconsole $(DESTDIR)$(PRIVATE_BINDIR)
+	$(INSTALL_PROG) xenconsole xenconsoled $(DESTDIR)$(PRIVATE_BINDIR)
 
 -include $(DEPS)
