--- a/tools/blktap/lib/Makefile
+++ b/tools/blktap/lib/Makefile
@@ -1,9 +1,7 @@
 XEN_ROOT = ../../..
 include $(XEN_ROOT)/tools/Rules.mk
 
-MAJOR    = 3.0
-MINOR    = 0
-SONAME   = libblktap.so.$(MAJOR)
+SONAME   = libblktap.so
 
 CFLAGS   += -I.
 CFLAGS   += $(CFLAGS_libxenctrl)
@@ -29,7 +27,7 @@
 IBINS   :=
 
 LIB_STATIC = libblktap.a
-LIB_SHARED = libblktap.so.$(MAJOR).$(MINOR)
+LIB_SHARED = libblktap.so
 
 .PHONY: all
 all: $(LIB_STATIC) $(LIB_SHARED)
@@ -41,19 +39,15 @@
 	$(INSTALL_DIR) $(DESTDIR)$(INCLUDEDIR)
 	$(INSTALL_DATA) $(LIB_STATIC) $(DESTDIR)$(LIBDIR)
 	$(INSTALL_DATA) $(LIB_SHARED) $(DESTDIR)$(PRIVATE_LIBDIR)
-	ln -sf libblktap.so.$(MAJOR).$(MINOR) $(DESTDIR)$(PRIVATE_LIBDIR)/libblktap.so.$(MAJOR)
-	ln -sf libblktap.so.$(MAJOR) $(DESTDIR)$(PRIVATE_LIBDIR)/libblktap.so
 	$(INSTALL_DATA) blktaplib.h $(DESTDIR)$(INCLUDEDIR)
 
 .PHONY: clean
 clean:
 	rm -rf *.a *.so* *.o *.opic *.rpm $(LIB) *~ $(DEPS) xen TAGS
 
-libblktap.so.$(MAJOR).$(MINOR): $(OBJS_PIC) 
-	$(CC) $(CFLAGS) -Wl,$(SONAME_LDFLAG) -Wl,$(SONAME) $(SHLIB_CFLAGS) \
+libblktap.so: $(OBJS_PIC)
+	$(CC) $(CFLAGS) $(SHLIB_CFLAGS) \
 	      $(LDFLAGS) -o $@ $^
-	ln -sf libblktap.so.$(MAJOR).$(MINOR) libblktap.so.$(MAJOR)
-	ln -sf libblktap.so.$(MAJOR) libblktap.so
 
 libblktap.a: $(OBJS) 
 	$(AR) rc $@ $^
