--- a/tools/firmware/etherboot/Makefile
+++ b/tools/firmware/etherboot/Makefile
@@ -14,50 +14,19 @@
 
 IPXE_TARBALL_URL := $(XEN_EXTFILES_URL)/ipxe-git-$(IPXE_GIT_TAG).tar.gz
 
-D=ipxe
-T=ipxe.tar.gz
-
-ROMS = $(addprefix $D/src/bin/, $(addsuffix .rom, $(NICS)))
+ROMS = $(addprefix /usr/lib/ipxe/, $(addsuffix .rom, $(NICS)))
 
 .NOTPARALLEL:
 
 .PHONY: all
 all: eb-roms.h
 
-%.rom: $D/src/arch/i386/Makefile
-	$(MAKE) -C $D/src bin/$(*F).rom
-
-eb-roms.h.new: $(ROMS)
+eb-roms.h: $(ROMS)
 	cat $^ | ../hvmloader/mkhex etherboot >$@
 
-eb-roms.h: Config
-	$(MAKE) NO_WERROR=1 $@.new
-	mv -f $@.new $@
-
-$T:
-	if ! wget -O _$T $(IPXE_TARBALL_URL); then \
-		$(GIT) clone $(IPXE_GIT_URL) $D.git; \
-		(cd $D.git && $(GIT) archive --format=tar --prefix=$D/ \
-		$(IPXE_GIT_TAG) | gzip >../_$T); \
-		rm -rf $D.git; \
-	fi
-	mv _$T $T
-
-$D/src/arch/i386/Makefile: $T Config
-	rm -rf $D
-	gzip -dc $T | tar xf -
-	for i in $$(cat patches/series) ; do                 \
-	    patch -d $D -p1 --quiet <patches/$$i || exit 1 ; \
-	done
-	cat Config >>$@
-
-$D/src/bin/NIC: $D/src/arch/i386/Makefile
-	$(MAKE) -C $D/src bin/NIC
-
 .PHONY: clean
 clean:
-	rm -rf $D $D.git *~ eb-roms.h _$T
+	rm -rf *~ eb-roms.h
 
 .PHONY: distclean
 distclean: clean
-	rm -rf $T
--- a/tools/firmware/etherboot/Config
+++ b/tools/firmware/etherboot/Config
@@ -1,5 +1,5 @@
 
-NICS = rtl8139 8086100e
+NICS = rtl8139 e1000_82540
 
 CFLAGS += -UPXE_DHCP_STRICT
 CFLAGS += -DPXE_DHCP_STRICT
