--- xen-unstable.orig/tools/libxc/Makefile
+++ xen-unstable/tools/libxc/Makefile
@@ -1,9 +1,6 @@
 XEN_ROOT = ../..
 include $(XEN_ROOT)/tools/Rules.mk
 
-MAJOR    = 3.2
-MINOR    = 0
-
 CTRL_SRCS-y       :=
 CTRL_SRCS-y       += xc_core.c
 CTRL_SRCS-$(CONFIG_X86) += xc_core_x86.c
@@ -74,12 +71,12 @@
 
 LIB := libxenctrl.a
 ifneq ($(stubdom),y)
-LIB += libxenctrl.so libxenctrl.so.$(MAJOR) libxenctrl.so.$(MAJOR).$(MINOR)
+LIB += libxenctrl.so
 endif
 
 LIB += libxenguest.a
 ifneq ($(stubdom),y)
-LIB += libxenguest.so libxenguest.so.$(MAJOR) libxenguest.so.$(MAJOR).$(MINOR)
+LIB += libxenguest.so
 endif
 
 .PHONY: all
@@ -96,16 +93,12 @@
 install: build
 	$(INSTALL_DIR) $(DESTDIR)$(PRIVATE_LIBDIR)
 	$(INSTALL_DIR) $(DESTDIR)$(INCLUDEDIR)
-	$(INSTALL_PROG) libxenctrl.so.$(MAJOR).$(MINOR) $(DESTDIR)$(PRIVATE_LIBDIR)
+	$(INSTALL_PROG) libxenctrl.so $(DESTDIR)$(PRIVATE_LIBDIR)
 	$(INSTALL_DATA) libxenctrl.a $(DESTDIR)$(PRIVATE_LIBDIR)
-	ln -sf libxenctrl.so.$(MAJOR).$(MINOR) $(DESTDIR)$(PRIVATE_LIBDIR)/libxenctrl.so.$(MAJOR)
-	ln -sf libxenctrl.so.$(MAJOR) $(DESTDIR)$(PRIVATE_LIBDIR)/libxenctrl.so
 	$(INSTALL_DATA) xenctrl.h $(DESTDIR)$(INCLUDEDIR)
 
-	$(INSTALL_PROG) libxenguest.so.$(MAJOR).$(MINOR) $(DESTDIR)$(PRIVATE_LIBDIR)
+	$(INSTALL_PROG) libxenguest.so $(DESTDIR)$(PRIVATE_LIBDIR)
 	$(INSTALL_DATA) libxenguest.a $(DESTDIR)$(PRIVATE_LIBDIR)
-	ln -sf libxenguest.so.$(MAJOR).$(MINOR) $(DESTDIR)$(PRIVATE_LIBDIR)/libxenguest.so.$(MAJOR)
-	ln -sf libxenguest.so.$(MAJOR) $(DESTDIR)$(PRIVATE_LIBDIR)/libxenguest.so
 	$(INSTALL_DATA) xenguest.h $(DESTDIR)$(INCLUDEDIR)
 
 .PHONY: TAGS
@@ -133,26 +126,16 @@
 libxenctrl.a: $(CTRL_LIB_OBJS)
 	$(AR) rc $@ $^
 
-libxenctrl.so: libxenctrl.so.$(MAJOR)
-	ln -sf $< $@
-libxenctrl.so.$(MAJOR): libxenctrl.so.$(MAJOR).$(MINOR)
-	ln -sf $< $@
-
-libxenctrl.so.$(MAJOR).$(MINOR): $(CTRL_PIC_OBJS)
-	$(CC) $(CFLAGS) $(LDFLAGS) -Wl,$(SONAME_LDFLAG) -Wl,libxenctrl.so.$(MAJOR) $(SHLIB_CFLAGS) $(call LDFLAGS_RPATH) -o $@ $^ $(PTHREAD_LIBS)
+libxenctrl.so: $(CTRL_PIC_OBJS)
+	$(CC) $(CFLAGS) $(LDFLAGS) $(SHLIB_CFLAGS) $(call LDFLAGS_RPATH) -o $@ $^ $(PTHREAD_LIBS)
 
 # libxenguest
 
 libxenguest.a: $(GUEST_LIB_OBJS)
 	$(AR) rc $@ $^
 
-libxenguest.so: libxenguest.so.$(MAJOR)
-	ln -sf $< $@
-libxenguest.so.$(MAJOR): libxenguest.so.$(MAJOR).$(MINOR)
-	ln -sf $< $@
-
-libxenguest.so.$(MAJOR).$(MINOR): $(GUEST_PIC_OBJS) libxenctrl.so
-	$(CC) $(CFLAGS) $(LDFLAGS) -Wl,$(SONAME_LDFLAG) -Wl,libxenguest.so.$(MAJOR) $(SHLIB_CFLAGS) $(call LDFLAGS_RPATH) -o $@ $(GUEST_PIC_OBJS) -lz -lxenctrl $(PTHREAD_LIBS)
+libxenguest.so: $(GUEST_PIC_OBJS) libxenctrl.so
+	$(CC) $(CFLAGS) $(LDFLAGS) $(SHLIB_CFLAGS) $(call LDFLAGS_RPATH) -o $@ $(GUEST_PIC_OBJS) -lz -lxenctrl $(PTHREAD_LIBS)
 
 -include $(DEPS)
 
--- xen-unstable.orig/tools/blktap/lib/Makefile
+++ xen-unstable/tools/blktap/lib/Makefile
@@ -1,9 +1,7 @@
 XEN_ROOT = ../../..
 include $(XEN_ROOT)/tools/Rules.mk
 
-MAJOR    = 3.0
-MINOR    = 0
-SONAME   = libblktap.so.$(MAJOR)
+SONAME   = libblktap.so
 
 CFLAGS   += -I.
 CFLAGS   += $(CFLAGS_libxenctrl)
@@ -28,7 +26,7 @@
 OBJS_PIC = $(SRCS:.c=.opic)
 IBINS   :=
 
-LIB      = libblktap.a libblktap.so.$(MAJOR).$(MINOR)
+LIB      = libblktap.a libblktap.so
 
 .PHONY: all
 all: $(LIB)
@@ -38,19 +36,15 @@
 	$(INSTALL_DIR) $(DESTDIR)$(PRIVATE_LIBDIR)
 	$(INSTALL_DIR) $(DESTDIR)$(INCLUDEDIR)
 	$(INSTALL_DATA) $(LIB) $(DESTDIR)$(PRIVATE_LIBDIR)
-	ln -sf libblktap.so.$(MAJOR).$(MINOR) $(DESTDIR)$(PRIVATE_LIBDIR)/libblktap.so.$(MAJOR)
-	ln -sf libblktap.so.$(MAJOR) $(DESTDIR)$(PRIVATE_LIBDIR)/libblktap.so
 	$(INSTALL_DATA) blktaplib.h $(DESTDIR)$(INCLUDEDIR)
 
 .PHONY: clean
 clean:
 	rm -rf *.a *.so* *.o *.opic *.rpm $(LIB) *~ $(DEPS) xen TAGS
 
-libblktap.so.$(MAJOR).$(MINOR): $(OBJS_PIC) 
-	$(CC) $(CFLAGS) -Wl,$(SONAME_LDFLAG) -Wl,$(SONAME) $(SHLIB_CFLAGS) \
+libblktap.so: $(OBJS_PIC) 
+	$(CC) $(CFLAGS) $(SHLIB_CFLAGS) \
 	      $(LDFLAGS) -o $@ $^
-	ln -sf libblktap.so.$(MAJOR).$(MINOR) libblktap.so.$(MAJOR)
-	ln -sf libblktap.so.$(MAJOR) libblktap.so
 
 libblktap.a: $(OBJS) 
 	$(AR) rc $@ $^
--- xen-unstable.orig/tools/libfsimage/common/Makefile
+++ xen-unstable/tools/libfsimage/common/Makefile
@@ -1,9 +1,6 @@
 XEN_ROOT = ../../..
 include $(XEN_ROOT)/tools/Rules.mk
 
-MAJOR = 1.0
-MINOR = 0
-
 CFLAGS += -Werror -Wp,-MD,.$(@F).d
 DEPS = .*.d
 
@@ -15,7 +12,7 @@
 
 PIC_OBJS := $(patsubst %.c,%.opic,$(LIB_SRCS-y))
 
-LIB = libfsimage.so libfsimage.so.$(MAJOR) libfsimage.so.$(MAJOR).$(MINOR)
+LIB = libfsimage.so
 
 .PHONY: all
 all: $(LIB)
@@ -24,9 +21,7 @@
 install: all
 	$(INSTALL_DIR) $(DESTDIR)$(PRIVATE_LIBDIR)
 	$(INSTALL_DIR) $(DESTDIR)$(INCLUDEDIR)
-	$(INSTALL_PROG) libfsimage.so.$(MAJOR).$(MINOR) $(DESTDIR)$(PRIVATE_LIBDIR)
-	ln -sf libfsimage.so.$(MAJOR).$(MINOR) $(DESTDIR)$(PRIVATE_LIBDIR)/libfsimage.so.$(MAJOR)
-	ln -sf libfsimage.so.$(MAJOR) $(DESTDIR)$(PRIVATE_LIBDIR)/libfsimage.so
+	$(INSTALL_PROG) libfsimage.so $(DESTDIR)$(PRIVATE_LIBDIR)
 	$(INSTALL_DATA) fsimage.h $(DESTDIR)$(INCLUDEDIR)
 	$(INSTALL_DATA) fsimage_plugin.h $(DESTDIR)$(INCLUDEDIR)
 	$(INSTALL_DATA) fsimage_grub.h $(DESTDIR)$(INCLUDEDIR)
@@ -34,13 +29,8 @@
 clean distclean:
 	rm -f $(PIC_OBJS) $(LIB) $(DEPS)
 
-libfsimage.so: libfsimage.so.$(MAJOR)
-	ln -sf $< $@
-libfsimage.so.$(MAJOR): libfsimage.so.$(MAJOR).$(MINOR)
-	ln -sf $< $@
-
-libfsimage.so.$(MAJOR).$(MINOR): $(PIC_OBJS)
-	$(CC) $(CFLAGS) $(LDFLAGS) -Wl,$(SONAME_LDFLAG) -Wl,libfsimage.so.$(MAJOR) $(SHLIB_CFLAGS) -o $@ $^ -lpthread
+libfsimage.so: $(PIC_OBJS)
+	$(CC) $(CFLAGS) $(LDFLAGS) $(SHLIB_CFLAGS) -o $@ $^ -lpthread
 
 -include $(DEPS)
 
--- xen-unstable.orig/tools/flask/libflask/Makefile
+++ xen-unstable/tools/flask/libflask/Makefile
@@ -1,6 +1,3 @@
-MAJOR    = 1.0
-MINOR    = 0
-
 XEN_ROOT = ../../..
 include $(XEN_ROOT)/tools/Rules.mk
 
@@ -22,7 +19,7 @@
 PIC_OBJS := $(patsubst %.c,%.opic,$(SRCS))
 
 LIB := libflask.a
-LIB += libflask.so libflask.so.$(MAJOR) libflask.so.$(MAJOR).$(MINOR)
+LIB += libflask.so
 
 .PHONY: all
 all: build
@@ -35,10 +32,8 @@
 install: build
 	$(INSTALL_DIR) $(DESTDIR)$(PRIVATE_LIBDIR)
 	$(INSTALL_DIR) $(DESTDIR)$(INCLUDEDIR)
-	$(INSTALL_PROG) libflask.so.$(MAJOR).$(MINOR) $(DESTDIR)$(PRIVATE_LIBDIR)
+	$(INSTALL_PROG) libflask.so $(DESTDIR)$(PRIVATE_LIBDIR)
 	$(INSTALL_DATA) libflask.a $(DESTDIR)$(PRIVATE_LIBDIR)
-	ln -sf libflask.so.$(MAJOR).$(MINOR) $(DESTDIR)$(PRIVATE_LIBDIR)/libflask.so.$(MAJOR)
-	ln -sf libflask.so.$(MAJOR) $(DESTDIR)$(PRIVATE_LIBDIR)/libflask.so
 	$(INSTALL_DATA) include/flask.h $(DESTDIR)$(INCLUDEDIR)
 
 .PHONY: TAGS
@@ -54,12 +49,7 @@
 libflask.a: $(LIB_OBJS)
 	$(AR) rc $@ $^
 
-libflask.so: libflask.so.$(MAJOR)
-	ln -sf $< $@
-libflask.so.$(MAJOR): libflask.so.$(MAJOR).$(MINOR)
-	ln -sf $< $@
-
-libflask.so.$(MAJOR).$(MINOR): $(PIC_OBJS)
-	$(CC) $(CFLAGS) $(LDFLAGS) -Wl,$(SONAME_LDFLAG) -Wl,libflask.so.$(MAJOR) $(SHLIB_CFLAGS) -o $@ $^
+libflask.so: $(PIC_OBJS)
+	$(CC) $(CFLAGS) $(LDFLAGS) $(SHLIB_CFLAGS) -o $@ $^
 
 -include $(DEPS)
--- xen-unstable.orig/tools/xenstore/Makefile
+++ xen-unstable/tools/xenstore/Makefile
@@ -1,9 +1,6 @@
 XEN_ROOT=../..
 include $(XEN_ROOT)/tools/Rules.mk
 
-MAJOR = 3.0
-MINOR = 0
-
 CFLAGS += -Werror
 CFLAGS += -I.
 CFLAGS += $(CFLAGS_libxenctrl)
@@ -66,15 +63,10 @@
 xs_tdb_dump: xs_tdb_dump.o utils.o tdb.o talloc.o
 	$(CC) $(CFLAGS) $(LDFLAGS) $^ -o $@
 
-libxenstore.so: libxenstore.so.$(MAJOR)
-	ln -sf $< $@
-libxenstore.so.$(MAJOR): libxenstore.so.$(MAJOR).$(MINOR)
-	ln -sf $< $@
-
 xs.opic: CFLAGS += -DUSE_PTHREAD
 
-libxenstore.so.$(MAJOR).$(MINOR): xs.opic xs_lib.opic
-	$(CC) $(CFLAGS) $(LDFLAGS) -Wl,$(SONAME_LDFLAG) -Wl,libxenstore.so.$(MAJOR) $(SHLIB_CFLAGS) -o $@ $^ $(SOCKET_LIBS) -lpthread
+libxenstore.so: xs.opic xs_lib.opic
+	$(CC) $(CFLAGS) $(LDFLAGS) $(SHLIB_CFLAGS) -o $@ $^ $(SOCKET_LIBS) -lpthread
 
 libxenstore.a: xs.o xs_lib.o
 	$(AR) rcs $@ $^
@@ -108,9 +100,7 @@
 		ln -f $(DESTDIR)$(PRIVATE_BINDIR)/xenstore $(DESTDIR)$(PRIVATE_BINDIR)/$${c} ; \
 	done
 	$(INSTALL_DIR) $(DESTDIR)$(PRIVATE_LIBDIR)
-	$(INSTALL_PROG) libxenstore.so.$(MAJOR).$(MINOR) $(DESTDIR)$(PRIVATE_LIBDIR)
-	ln -sf libxenstore.so.$(MAJOR).$(MINOR) $(DESTDIR)$(PRIVATE_LIBDIR)/libxenstore.so.$(MAJOR)
-	ln -sf libxenstore.so.$(MAJOR) $(DESTDIR)$(PRIVATE_LIBDIR)/libxenstore.so
+	$(INSTALL_PROG) libxenstore.so $(DESTDIR)$(PRIVATE_LIBDIR)
 	$(INSTALL_DATA) libxenstore.a $(DESTDIR)$(PRIVATE_LIBDIR)
 	$(INSTALL_DATA) xs.h $(DESTDIR)$(INCLUDEDIR)
 	$(INSTALL_DATA) xs_lib.h $(DESTDIR)$(INCLUDEDIR)
