--- xen-unstable.orig/tools/libfsimage/Rules.mk
+++ xen-unstable/tools/libfsimage/Rules.mk
@@ -7,11 +7,7 @@
 
 PIC_OBJS := $(patsubst %.c,%.opic,$(LIB_SRCS-y))
 
-FSDIR-$(CONFIG_Linux) = $(LIBDIR)/fs/$(FS)
-FSDIR-$(CONFIG_SunOS)-x86_64 = $(PREFIX)/lib/fs/$(FS)/64
-FSDIR-$(CONFIG_SunOS)-x86_32 = $(PREFIX)/lib/fs/$(FS)/
-FSDIR-$(CONFIG_SunOS) = $(FSDIR-$(CONFIG_SunOS)-$(XEN_TARGET_ARCH))
-FSDIR = $(FSDIR-y)
+FSDIR = $(PRIVATE_LIBDIR)/fs/$(FS)
 
 FSLIB = fsimage.so
 
--- xen-unstable.orig/tools/libfsimage/common/Makefile
+++ xen-unstable/tools/libfsimage/common/Makefile
@@ -1,7 +1,7 @@
 XEN_ROOT = ../../..
 include $(XEN_ROOT)/tools/Rules.mk
 
-CFLAGS += -Werror -Wp,-MD,.$(@F).d
+CFLAGS += -Werror -Wp,-MD,.$(@F).d -DLIBDIR="\"$(LIBDIR)\""
 DEPS = .*.d
 
 LDFLAGS-$(CONFIG_SunOS) = -Wl,-M -Wl,mapfile-SunOS
--- xen-unstable.orig/tools/libfsimage/common/fsimage_plugin.c
+++ xen-unstable/tools/libfsimage/common/fsimage_plugin.c
@@ -131,23 +131,8 @@
 	int err;
 	int ret = -1;
 
-#ifdef __sun__
 	if (fsdir == NULL)
-		fsdir = "/usr/lib/fs";
-
-	if (sizeof(void *) == 8)
-		isadir = "64/";
-#elif defined(__ia64__)
-	if (fsdir == NULL)
-		fsdir = "/usr/lib/fs";
-#else
-	if (fsdir == NULL) {
-		if (sizeof(void *) == 8)
-			fsdir = "/usr/lib64/fs";
-		else
-			fsdir = "/usr/lib/fs";
-	}
-#endif
+		fsdir = "/usr/" LIBDIR "/fs";
 
 	if ((name_max = pathconf(fsdir, _PC_NAME_MAX)) == -1)
 		goto fail;
